### PR TITLE
rename some long documentation files and add more logging to rename command.

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose
https://jira.autodesk.com/browse/DYN-5508

TODO:
- [x] file followups

some doc files have names that are too long - this PR does the following:
* renames the known ones that have caused issues for host installers.
* as a byproduct - renames MOST of the related support files. (images/dyns)
* keeps the existing files - these will still need to be excluded by some hosts as other hosts have included them.
* renames all overloads with the same names as these nodes.
* adds a verbose options to the rename command - this will log some warnings and generate a rename log, which I have committed - if we continue to use this rename process, I think we should append any new logs to the existing one - the command does this already if a log is found.

an example of the kind of warning that will be generated:

```
PS C:\Users\kirschm> cd C:\Users\kirschm\source\repos\mjkkirschner\Dynamo\src\Tools\NodeDocumentationMarkdownGenerator\bin\Debug
PS C:\Users\kirschm\source\repos\mjkkirschner\Dynamo\src\Tools\NodeDocumentationMarkdownGenerator\bin\Debug> .\NodeDocumentationMarkdownGenerator.exe rename -d "C:\Users\kirschm\source\repos\mjkkirschner\Dynamo\doc\distrib\NodeHelpFiles\to rename" -v -m 10

 _______
/       \
$$$$$$$  | __    __  _______    ______   _____  ____    ______
$$ |  $$ |/  |  /  |/       \  /      \ /     \/    \  /      \
$$ |  $$ |$$ |  $$ |$$$$$$$  | $$$$$$  |$$$$$$ $$$$  |/$$$$$$  |
$$ |  $$ |$$ |  $$ |$$ |  $$ | /    $$ |$$ | $$ | $$ |$$ |  $$ |
$$ |__$$ |$$ \__$$ |$$ |  $$ |/$$$$$$$ |$$ | $$ | $$ |$$ \__$$ |
$$    $$/ $$    $$ |$$ |  $$ |$$    $$ |$$ | $$ | $$ |$$    $$/
$$$$$$$/   $$$$$$$ |$$/   $$/  $$$$$$$/ $$/  $$/  $$/  $$$$$$/
          /  \__$$ |
          $$    $$/
           $$$$$$/

looking for dynamo core assemblies in ..\..\..\..\..\bin\AnyCPU\Debug
Autodesk.DesignScript.Geometry.CoordinateSystem.Transform(coordinateSystem, fromCoordinateSystem, contextCoordinateSystem).md references Autodesk.DesignScript.Geometry.CoordinateSystem.Transform(CS, fromCS, contextCS)_img.png, but C:\Users\kirschm\source\repos\mjkkirschner\Dynamo\doc\distrib\NodeHelpFiles\to rename\Autodesk.DesignScript.Geometry.CoordinateSystem.Transform(CS, fromCS, contextCS)_img.png will not be renamed as it does not contain the same basename as the md file. Manually rename this file and the reference.
Autodesk.DesignScript.Geometry.CoordinateSystem.Transform(CS, fromCS, contextCS)_img.png was not found in the rename log
```

See `rename_log.txt` for the log.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [x] This PR contains no files larger than 50 MB 

### Release Notes

Rename some documentation files so they can be shipped with dynamo.

### CODE REVIEW

**Comment by pinzart90** [comment]
not sure I understand this fully:
```
renames the known ones that have caused issues for host installers.
keeps the existing files - these will still need to be excluded by some hosts as other hosts have included them.
```
So you renamed some content (.dyn) from some long name to a GUID based name. These renamed files have been added to git.
The old (long named) files are still in git (not removed) because host integrators still expect them ? So now we have duplicate content ?

File: src/Tools/NodeDocumentationMarkdownGenerator/Commands/RenameCommand.cs [hunk 1]
```diff
@@ -75,6 +103,29 @@ private static void RenameDirectory(string directory, int maxLength)
             }
 
             var allMdFiles = Directory.GetFiles(directory, "*.md", SearchOption.TopDirectoryOnly).Select(x => new FileInfo(x)).ToList();
+            var allFiles = Directory.GetFiles(directory,"*",SearchOption.TopDirectoryOnly).Select(x => new FileInfo(x)).ToList();
+            var allNonMDFiles = allFiles.Except(allMdFiles);
+            if (Verbose)
+            {
+                //do a scan for support files which do not share the base name.
+                foreach (var nonMDFile in allNonMDFiles)
```
Comment (sm6srw) [inline src/Tools/NodeDocumentationMarkdownGenerator/Commands/RenameCommand.cs:111]: So you did find support files without a base file? What that caused by someone deleting the basename but not the supportfiles? I can't remember seeing this before.

File: src/Tools/NodeDocumentationMarkdownGenerator/Commands/RenameCommand.cs [hunk 2]
```diff
@@ -1,14 +1,29 @@
...
     {
+        /// <summary>
+        /// simple output of rename operations that were executed during the command.
+        /// </summary>
+        internal static List<string> Log { get; set; } = new List<string>();
+        internal static bool Verbose { get; set; }
         internal static void HandleRename(RenameOptions opts)
         {
+            
+            Verbose=opts.Verbose;
```
Comment (sm6srw) [inline src/Tools/NodeDocumentationMarkdownGenerator/Commands/RenameCommand.cs:22]: Thanks for adding Verbose support.

File: src/Tools/NodeDocumentationMarkdownGenerator/Commands/RenameCommand.cs [hunk 3]
```diff
@@ -75,6 +103,29 @@ private static void RenameDirectory(string directory, int maxLength)
...
+            var allFiles = Directory.GetFiles(directory,"*",SearchOption.TopDirectoryOnly).Select(x => new FileInfo(x)).ToList();
+            var allNonMDFiles = allFiles.Except(allMdFiles);
+            if (Verbose)
+            {
+                //do a scan for support files which do not share the base name.
+                foreach (var nonMDFile in allNonMDFiles)
+                {
+                    foreach (var mdfile in allMdFiles)
+                    {
+                        //md file contains a reference to the support file, but the names are not the same.
```
Comment (sm6srw) [inline src/Tools/NodeDocumentationMarkdownGenerator/Commands/RenameCommand.cs:115]: Nice consistency check. I don't remember seeing that case neither. But it was much fewer files when i did this.

**Review by sm6srw — APPROVED** [review]
LGTM!

File: src/Tools/NodeDocumentationMarkdownGenerator/Commands/RenameCommand.cs (hunk 1 above)
Comment (mjkkirschner) [reply src/Tools/NodeDocumentationMarkdownGenerator/Commands/RenameCommand.cs:111]: there is a single support file that is referenced from one of the renamed files, but its argument names are different, so it does not get renamed.

See my sample output for the example.

I didn't end up changing it as the name seemed short enough.

**Comment by mjkkirschner** [comment]
thanks for the review- I will merge after the tests pass.
will file two followups
1. remove duplicate files for GL25
2. spike on delivering builtin node docs in resource assembly to avoid long path issue.

**Pull request merged**